x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/rare_job_creator.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedSearchSavedObject } from '../../../../../../common/types/kibana';
+import { JobCreator } from './job_creator';
+import { Field, SplitField } from '../../../../../../common/types/fields';
+import { Job, Datafeed } from '../../../../../../common/types/anomaly_detection_jobs';
+import { JOB_TYPE, CREATED_BY_LABEL } from '../../../../../../common/constants/new_job';
+import { getRichDetectors } from './util/general';
+import { IndexPattern } from '../../../../../../../../../src/plugins/data/public';
+import { isSparseDataJob } from './util/general';
+import { ML_JOB_AGGREGATION } from '../../../../../../common/constants/aggregation_types';
+
+export class RareJobCreator extends JobCreator {
+  private _rareField: Field | null = null;
+  private _populationField: SplitField = null;
+  private _splitField: SplitField = null;
```

Comment:
Do many `rare` jobs used a partition field? Just wondering if this is something we could leave for the advanced wizard?

Reply:
A significant proportion of `rare` jobs will use a partition field, so definitely worth keeping this in the wizard.